src/appleseed/renderer/kernel/shading/closures.cpp

```diff
@@ -1167,6 +1167,13 @@ void CompositeClosure::compute_cdf()
     const size_t closure_count = get_num_closures();
     if (closure_count > 0)
     {
+        if (closure_count == 1)
```

Comment:
What's the most common case? I suggest rearranging the tests to minimize the number of tests in the common case.